input/drivers/udev_input.c

```diff
@@ -3970,6 +4096,8 @@ static bool udev_set_sensor_state(void *data, unsigned port, enum retro_sensor_a
          /* If already disabled, then do nothing */
          linux_close_illuminance_sensor(udev->illuminance_sensor); /* noop if NULL */
          udev->illuminance_sensor = NULL;
+      case RETRO_SENSOR_GYROSCOPE_ENABLE:
+      case RETRO_SENSOR_ACCELEROMETER_ENABLE:
```

Comment:
Did you forget to implement these?

Reply:
There is no need to

Reply:
Isn't there overhead in polling sensors and running the callbacks? Maybe it's not big, but cores that don't use it shouldn't have to pay for it. I would prefer it if you only opened and polled the sensors that the core actually asks for.

Reply:
But `udev` itself is still polling the sensors and running the callback, right?

Reply:
Ah I see... Will fix

Reply:
open_devices automatically enables everything so I can't really remove the callback initialization.

Reply:
What makes you think so? Is there any reason you can't hold off on opening a sensor until the core asks for one?

Reply:
There's already code that assumes everything will be initialized and enabled in one go in a big for loop. If I don't initialize and open the sensor retroarch won't know what sensors are even available or even if sensors are available.

Reply:
Actually I might be able to rig something one moment

Reply:
Nope, sorry. `udev_input_add_device` just does too much. And I really don't want to make this pull request have another big refactor that will take a month to write and 6 months to verify.

Reply:
> But `udev` itself is still polling the sensors and running the callback, right?

udev will keep polling the sensors regardless of if we want to or not. And the "callback" is actually a file read operation. `udev_input_poll` already just reads all the open file descriptors regardless of whether they are requested or not. If you want to optimize the udev input driver go ahead but I think this is beyond the scope of this pull request.

Reply:
Honestly I think either way the branching to check whether the input was requested or not will be more expensive than just running it unconditionally.